R/install-dev.R

```diff
@@ -28,15 +28,7 @@ install_dev <- function(package, cran_url = getOption("repos")[["CRAN"]], ...) {
     cran_url <- "https://cloud.r-project.org"
   }
 
-  # if @[ref] is included, save for repo install
-  if (contains_ref(package)) {
-    ref <- sub(".*(?=(@|#))", "", package, perl = TRUE)
-    package <- sub("(#|@).*", "", package)
-  } else {
-    ref <- ""
-  }
-
-  url <- build_url(cran_url, "web", "packages", package, "DESCRIPTION")
+  url <- build_url(cran_url, "web", "packages", dev_split_ref(package)[["pkg"]], "DESCRIPTION")
```

Comment:
I don't know if it's the best way to access "pkg" and "ref" values, but I used double brackets and column names.

Reply:
Yes, that's good, but can we call `dev_split_ref` only once? Not a big a deal performance-wise, more of a principle.  Store the result here, and use it later below.